src/stacks-hierarchy/utils.ts

```diff
@@ -69,3 +69,17 @@ export const useDefaultTariffZone = (
     return {...tariffZones[0], resultType: 'zone'};
   }, [tariffZones, tariffZoneFromLocation]);
 };
+
+export const useFilterTariffZone = (
+  tariffZones: TariffZone[],
+  allowedTariffZones: string[],
```

Comment:
I would prefer if `allowedTariffZones` had suffix "refs" or "ids". Based on the naming `tariffZones` and `allowedTariffZones` both seems to be an array of `TariffZone`.

Reply:
done :D